Source/ARTAuth.m

```diff
@@ -102,6 +108,11 @@ - (ARTTokenParams *)mergeParams:(ARTTokenParams *)customParams {
     return customParams ? customParams : [[ARTTokenParams alloc] initWithOptions:self.options];
 }
 
+- (ARTTokenParams *)replaceParams:(ARTTokenParams *)customParams {
+    ARTTokenParams* tkp = [[ARTTokenParams alloc] initWithOptions: (ARTClientOptions*)_replacedAuthOptions];
```

Comment:
This isn't a type-safe cast. Either ARTTokenParams needs a AuthOptions and not a full ClientOptions, in which case its constructor should be changed to reflect that, or it does need a full ClientOptions and that's what we should be giving it. (I suspect it's the former.)